internal/leakcheck/leakcheck_test.go

```diff
@@ -47,16 +51,14 @@ func TestCheck(t *testing.T) {
 	for i := 0; i < leakCount; i++ {
 		go func() { time.Sleep(2 * time.Second) }()
 	}
-	if ig := interestingGoroutines(); len(ig) == 0 {
-		t.Error("blah")
+	if leaked := interestingGoroutines(); len(leaked) != leakCount {
+		t.Errorf("interestingGoroutines found %v leaks, want %v leaks", len(leaked), leakCount)
 	}
 	e := &testLogger{}
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 	defer cancel()
-	CheckGoroutines(ctx, e)
-	if e.errorCount != leakCount {
+	if CheckGoroutines(ctx, e); e.errorCount == 0 {
```

Comment:
That works, or you can do

```suggestion
	if CheckGoroutines(ctx, e); ctx.Err() == nil {
```

Or even:

```suggestion
	if CheckGoroutines(ctx, e); e.errorCount < 3 {
```

Reply:
This check expects that the goroutines spawned by time.Sleep() should still be alive.
I have not put `e.errorcount == 3`, because ctx can cause its goroutine live longer and it will cause a errorcount of 4 (Because of this the test was flaky)

Reply:
> I have not put e.errorcount == 3, because ctx can cause its goroutine live longer and it will cause a errorcount of 4

Right, which is why I suggested changing to failing if it's less than three.  We should have at least three.  Or we can just confirm the check failed by checking `ctx.Err()` and then not have to worry about all the complexity to track `errorCount` at all.  Either way is fine.

Reply:
> Right, which is why I suggested changing to failing if it's less than three. We should have at least three. Or we can just confirm the check failed by checking `ctx.Err()` and then not have to worry about all the complexity to track `errorCount` at all. Either way is fine.

Right, make sense. Both works fine. I'm going with `e.errorcount < 3`for now.